utilities.py

```diff
@@ -85,13 +85,62 @@ def read_file(self):
         return headers, table
 
 
-# TODO Part 5: Implement the conversion from Quaternion to Euler Angles
+# Part 5: Implement the conversion from Quaternion to Euler Angles
 def euler_from_quaternion(quat):
     """
     Convert quaternion (w in last place) to euler roll, pitch, yaw.
     quat = [x, y, z, w]
     """
-    ... # just unpack yaw
-    return yaw
 
+    x, y, z, w = normalize_quaternion(quat)
+
+    # Note that we are choosing not to handle gimbal lock because it is very unlikely to happen in our case
+    # the pitch shouldn't be changing much
+    # If needed, we could test for gimbal lock
+
+    # Calculate Roll - commented as not used
+    # roll = calculate_roll(x, y, z, w)
+
+    # Calculate Pitch - commented as not used
+    # pitch = calculate_pitch(x, y, z, w)
+
+    # Calculate Yaw
+    yaw = calculate_yaw(x, y, z, w)
+
+    # return yaw in degrees (can be changed to radians if wanted)
+    return yaw * 180 / M_PI
```

Comment:
I suggest returning in radians since matlab and numpy likes radians.

Also maybe do the following:
`return roll_x, pitch_y, yaw_z`
Such that this can be used for all quaternions (This will require you to change the function usages in motion.py)